main.tf

```diff
@@ -157,7 +157,7 @@ module "eks" {
   create_bastion_sg            = var.bastion != null
   bastion_security_group_id    = try(module.bastion[0].security_group_id, null)
   bastion_public_ip            = try(module.bastion[0].public_ip, null)
-  bastion_user                 = try(var.bastion.username, null)
+  bastion_user                 = try(module.bastion[0].user, null)
   kubeconfig_path              = local.kubeconfig_path
   node_groups                  = local.node_groups
   node_iam_policies            = module.storage.iam_policies
```

Comment:
Not the actual place for the comment, but does `module "bastion'` need a reference to `var.bastion.user`?

Reply:
bastion takes the user 
```hcl
variable "bastion_user" {
  type        = string
  description = "ec2 instance user."
  default     = "ec2-user"

}
```

I think is the bastion module what should determine what the default user should be, so we should just source it

Reply:
This goes back to my comment where we should be uniform in the structure of the vars.

Reply:
I don't think we're passing the variable from `var.bastion` at the top-level is all:
```
diff --git a/main.tf b/main.tf
index 5291ef3..d72daad 100644
--- a/main.tf
+++ b/main.tf
@@ -113,6 +113,7 @@ module "bastion" {
   instance_type     = var.bastion.instance_type
   kms_key           = local.kms_key_arn
   install_binaries  = var.bastion.install_binaries
+  bastion_user      = var.bastion.username
   k8s_version       = var.k8s_version
   security_group_rules = {
     bastion_inbound_ssh = {
```

Reply:
I still think this needs to be addressed ^